src/kebab.rs

```diff
@@ -46,4 +46,12 @@ mod tests {
     t!(test8: "this-contains_ ALLKinds OfWord_Boundaries" => "this-contains-all-kinds-of-word-boundaries");
     t!(test9: "XΣXΣ baﬄe" => "xσxς-baﬄe");
     t!(test10: "XMLHttpRequest" => "xml-http-request");
+    t!(test11: "ファイルを読み込み" => "ファイルを読み込み");
+    t!(test12: "お前はもう死んでいる！何？" => "お前はもう死んでいる何");
+    t!(test13: "石室诗士施氏，嗜狮，誓食十狮。" => "石室诗士施氏嗜狮誓食十狮");
+    t!(test14: "石室詩士施氏，嗜獅，誓食十獅。" => "石室詩士施氏嗜獅誓食十獅");
+    t!(test15: "ㄕˊㄕˋㄕㄕˋㄕㄕˋ，ㄕˋㄕ，ㄕˋㄕˊㄕˊㄕ。" => "ㄕˊㄕˋㄕㄕˋㄕㄕˋㄕˋㄕㄕˋㄕˊㄕˊㄕ");
+    t!(test16: "shí shì shī shì shī shì ， shì shī ， shì shí shí shī 。" => "shí-shì-shī-shì-shī-shì-shì-shī-shì-shí-shí-shī");
+    t!(test17: "sek6 sat1 si1 si6 si1 si6 ，si3 si1 ，sai6 sik6 sap6 si1 。" => "sek6-sat1-si1-si6-si1-si6-si3-si1-sai6-sik6-sap6-si1");
+    t!(test18: "唱K" => "唱-k"); // this is incorrect but it doesn't matter
```

Comment:
Why do you consider this incorrect, and why does it not matter?

Reply:
Because it mixes chinese and english characters, we need to process chucks of characters rather than each characters, that would blown the complexity, like checking if `helloworld` should be `hello world`. Most of the time using `-` here would be correct since chinese characters are usually separated from english characters, but in this case both the chinese character and the english character form a chinese word.

Reply:
To be honest, this makes me want to abandon CJK support entirely. Predictably bad results are one thing, then you can just choose to look for another solution or roll your own. Mostly good results with hard to explain edge cases that are not documented, easy to miss and not going to be fixed is really bad IMO.

Reply:
@trinity-1686a since you're the only one who has expressed a need for this, what is your take on this (and the rest of the discussion, if you have time to read it)?

Reply:
As stated on the original issue, I do not speak any of those languages, and my issue is a mere mirror of [the issue someone have with a software I used to participate in developing](https://github.com/Plume-org/Plume/issues/326). I'm acting mainly as a proxy, with not much of an opinion on a subject I have very few knowledge of.
That said, I believe the results given by this pr are usually better than master, and such occasions on which it output something slightly different than expected are both rare and not _that_ annoying. It won't cause crash or obviously wrong results, more like odd looking ones

Reply:
> To be honest, this makes me want to abandon the idea of CJK support entirely. Predictably bad results are one thing, then you can just choose to look for another solution or roll your own. Mostly good results with hard to explain edge cases that are not documented, easy to miss and not going to be fixed is really bad IMO.

I don't think we should abandon the support for less than 1% or maybe even 0.01% of edge cases. It is not possible to handle all of it, especially when the word itself is made of mixed language. English have tons of edge cases too, like "ballet" is pronounced as "balley", but if there are tons of these so we just abandoned english support? That's why in this case I believe these edge cases are okay since they are rare and too complex to handle.

>  That said, I believe the results given by this pr are usually better than master, and such occasions on which it output something slightly different than expected are both rare and not that annoying. It won't cause crash or obviously wrong results, more like odd looking ones

Yes, definitely https://baptiste.gelez.xyz/~/終章/白い悪魔が黒い天国と出会い歴史は紡がれる/ is better than https://baptiste.gelez.xyz/~/終章/白-い-悪-魔-が-黒-い-天-国-と-出-会-い-歴-史-は-紡-が-れ-る/, it's no difference than joining each characters with `-` library.

Reply:
@trinity-1686a your original PR includes https://en.wikipedia.org/wiki/Kanbun_(Unicode_block), is there even people using that? I believe it is covered by other blocks already right?

Reply:
> To be honest, this makes me want to abandon the idea of CJK support entirely. Predictably bad results are one thing, then you can just choose to look for another solution or roll your own. Mostly good results with hard to explain edge cases that are not documented, easy to miss and not going to be fixed is really bad IMO.

If that's the case, maybe we should drop english support too. Because our english support is also incorrect. "YouTube" gets turned into "you-tube" instead of "youtube" and the same goes for "DreamWorks".

Reply:
It's a fair point though. There is no way to make it 100% correct for any language besides having to hard-code a list of them and keeping them up-to-date. And I am sure non of us would want to do that :)

I do agree with @pickfire and their example one comment above: 
> Yes, definitely https://baptiste.gelez.xyz/~/終章/白い悪魔が黒い天国と出会い歴史は紡がれる/ is better than https://baptiste.gelez.xyz/~/終章/白-い-悪-魔-が-黒-い-天-国-と-出-会-い-歴-史-は-紡-が-れ-る/, it's no difference than joining each character with -.

It's not perfect, but it's definitely better than the current support on the main branch.

Reply:
See discussion below, outside of this thread. An alternative approach has been identified that would solve the issue better without having to hardcode a list of languages, but implementation is still outstanding. I'll resolve this thread since it's outdated.

Reply:
I missed that , sorry!